tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyTool.java

```diff
@@ -220,6 +220,28 @@ private void includeCss(HttpServletRequest request, PrintWriter out) {
                         + ".css' />\n");
     }
 
+    private static final String DD_CLIENT_TOKEN = System.getenv("DD_CLIENT_TOKEN");
+    private static final String DD_CLIENT_APPID = System.getenv("DD_CLIENT_APPID");
+    private static final String DD_GIT_COMMIT_SHA = System.getenv("DD_GIT_COMMIT_SHA");
+    private static final String DD_ENV = System.getProperty("dd.env", "");
+
+    /** if DD_CLIENT_TOKEN is set, set these variables so index.js can pick them up. */
+    public static void includeDatadog(Writer out) throws IOException {
+        if (DD_CLIENT_TOKEN != null && !DD_CLIENT_TOKEN.isEmpty()) {
+            out.write(
+                    String.format(
+                            "<script>\n"
+                                    + "window.dataDogClientToken='%s';\n"
+                                    + "window.dataDogAppId='%s';\n"
+                                    + "window.dataDogEnv='%s';\n"
+                                    + "window.dataDogSha='%s';\n"
+                                    + "</script>\n",
+                            DD_CLIENT_TOKEN, DD_CLIENT_APPID, DD_ENV, DD_GIT_COMMIT_SHA));
+        } else {
+            out.write("<script>window.dataDogClientToken='';</script>\n");
+        }
+    }
```

Comment:
@btangmu this might look familiar. Here, I need to get *runtime* config information (from the actual staging server), so the webpack process won't work here.

Reply:
right -- still it's a drag to have more js in html in java -- ideally the front end should get the data from the back end as json

could we add a request for that json in cldrRun.mjs instead of having this in the initial html?

Reply:
That adds another round trip.